espresso/submitter/three_polling_run_loops.go

```diff
@@ -0,0 +1,833 @@
+package submitter
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"sync"
+	"time"
+
+	espresso_client "github.com/EspressoSystems/espresso-network/sdks/go/client"
+	espresso_light_client "github.com/EspressoSystems/espresso-network/sdks/go/light-client"
+	tagged_base64 "github.com/EspressoSystems/espresso-network/sdks/go/tagged-base64"
+	espresso_types "github.com/EspressoSystems/espresso-network/sdks/go/types"
+	"github.com/ccoveille/go-safecast"
+	"github.com/hf/nitrite"
+	"github.com/hf/nsm"
+	"github.com/hf/nsm/request"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/arbutil"
+	espresso_key_manager "github.com/offchainlabs/nitro/espresso/key-manager"
+	"github.com/offchainlabs/nitro/util"
+	"github.com/offchainlabs/nitro/util/dbutil"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+var (
+	// Schema extensions for Espresso Tracking
+	espressoSubmittedTxns        []byte = []byte("_espressoSubmittedTxns")    // contains the hash and pos of the submitted transactions
+	espressoPendingTxnsPositions []byte = []byte("_espressoPendingTxnsPos")   // contains the index of the pending txns that need to be submitted to espresso
+	espressoLastConfirmedPos     []byte = []byte("_espressoLastConfirmedPos") // contains the position of the last confirmed message
+)
+
+// ThreePollingRunLoops is a struct that implements the EspressoSubmitter interface
+// and provides methods to submit transactions to Espresso, manage pending.
+type ThreePollingRunLoops struct {
```

Comment:
Can we change this name from `ThreePollingRunLoops` to something else? Like maybe `EspressoManagerStateTrackers`?

Reply:
Yes just looking for a better name than ThreePollingRunLoops, I am fine if we dont name it `EspressoManagerStateTrackers`. Lets just think of a more descriptive name